src/main/java/com/jadventure/game/entities/Entity.java

```diff
@@ -257,6 +239,29 @@ public String getWeapon() {
         return result;
     }
 
+    private EquipmentLocation checkNewEquipmentLocation(EquipmentLocation place, Item item) {
```

Comment:
As per my comment below, this method primarily handles the job of unequipping. So, I propose an alternate name I believe may be clearer (I've also added changes to address my other suggestion), 
```suggestion
    private unequipConflictingItems(EquipmentLocation place, Item item) {
```